### ISSUE TITLE
Use shell instead of split_command_line()

### ISSUE THREAD

**Issue by jdemeyer** [pexpect/pexpect#387]
The use of `split_command_line` is quite confusing: it ''almost'' behaves like running a command in a shell, but not quite. This is unexpected to me, even though it is documented:
```
        Remember that Pexpect does NOT interpret shell meta characters such as
        redirect, pipe, or wild cards (``>``, ``|``, or ``*``). This is a
        common mistake.  If you want to run a command and pipe it through
        another command then you must also start a shell. For example::
```
If too many people make a "common mistake", maybe it's because the interface is confusing? So, instead of trying to be clever with the `split_command_line` function, why not simply use a shell?

**Comment by takluyver** [comment]
If you want to run the command in a shell, you can do that. I don't think it's helpful to make that automatic at this point, because it changes the meaning of code that already works.

**Comment by jdemeyer** [comment]
> If you want to run the command in a shell, you can do that. I don't think it's helpful to make that automatic at this point, because it changes the meaning of code that already works.

Right, but would it really change the meaning in a noticable way? Which runtime behaviour would change just because there is shell running your command?

As an alternative: what about a keyword `shell=False` to `spawn()` and `run()`, similar to [subprocess.Popen](https://docs.python.org/2/library/subprocess.html#popen-constructor)? This way, you wouldn't manually need to write `bash -c "some_command"` which is awkward.

**Comment by takluyver** [comment]
I would be fine with adding a `shell` parameter to control that.
